### PR TITLE
Feature/calculate entitlement

### PR DESCRIPTION
**Pull request by YetAnotherMatt**

### CODE REVIEW

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/EntitlementCalculator.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.dhsc.htbhf.claimant.entity.Claimant;
+import uk.gov.dhsc.htbhf.claimant.model.eligibility.EligibilityResponse;
+
+@Component
+@Slf4j
+public class EntitlementCalculator {
```
Comment (mikeSmith08) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/EntitlementCalculator.java]: Can we have some documentation on the business rules being applied please, either here in JavaDoc or in the README.md?

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/EntitlementCalculator.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+        this.pregnancyEntitlementCalculator = pregnancyEntitlementCalculator;
+        this.vouchersForChildrenUnderOne = vouchersForChildrenUnderOne;
+        this.vouchersForChildrenBetweenOneAndFour = vouchersForChildrenBetweenOneAndFour;
+        this.vouchersForPregnancy = vouchersForPregnancy;
+    }
+
+    public int calculateVoucherEntitlement(Claimant claimant, EligibilityResponse eligibilityResponse) {
+
+        int numberOfChildrenUnderFour = eligibilityResponse.getNumberOfChildrenUnderFour() == null ? 0 : eligibilityResponse.getNumberOfChildrenUnderFour();
```
Comment (mikeSmith08) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/EntitlementCalculator.java]: How about using something like `ObjectUtils.defaultIfNull` or similar from Commons Lang? I guess that might not be ideal as you're dealing with primitives.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/PregnancyEntitlementCalculator.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+@Component
+public class PregnancyEntitlementCalculator {
+
+    private final int pregnancyGracePeriodInDays;
+
+    public PregnancyEntitlementCalculator(@Value("${entitlement.pregnancy-grace-period-in-days}") int pregnancyGracePeriodInDays) {
+        this.pregnancyGracePeriodInDays = pregnancyGracePeriodInDays;
+    }
+
+    public boolean isEntitledToVoucher(LocalDate dueDate) {
```
Comment (mikeSmith08) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/PregnancyEntitlementCalculator.java]: Some docs here too would be helpful...

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/PregnancyEntitlementCalculatorTest.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+class PregnancyEntitlementCalculatorTest {
+
+    private static final int PREGNANCY_GRACE_PERIOD_IN_DAYS = 14;
+
+    PregnancyEntitlementCalculator calculator;
+
+
+    @BeforeEach
+    void setup() {
+        calculator = new PregnancyEntitlementCalculator(PREGNANCY_GRACE_PERIOD_IN_DAYS);
```
Comment (dwybourn) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/PregnancyEntitlementCalculatorTest.java]: This line could be moved to where we declare the calculator. Less code and it means we don't create a new object with every test.

**Pull request merged**